crates/router/src/connector/trustpay.rs

```diff
@@ -587,23 +589,122 @@ impl ConnectorIntegration<api::RSync, types::RefundsData, types::RefundsResponse
 impl api::IncomingWebhook for Trustpay {
     fn get_webhook_object_reference_id(
         &self,
-        _request: &api::IncomingWebhookRequestDetails<'_>,
-    ) -> CustomResult<String, errors::ConnectorError> {
-        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<ObjectReferenceId, errors::ConnectorError> {
+        let details: trustpay::TrustpayWebhookResponse = request
+            .body
+            .parse_struct("TrustpayWebhookResponse")
+            .change_context(errors::ConnectorError::WebhookBodyDecodingFailed)?;
+        match details.payment_information.credit_debit_indicator {
+            trustpay::CreditDebitIndicator::Crdt => Ok(ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::PaymentIntentId(
+                    details.payment_information.references.merchant_reference,
+                ),
+            )),
+            _ => Ok(ObjectReferenceId::RefundId(
```

Comment:
Why all default cases are going to refund case?.

Reply:
CRDT -> referes to Transaction events
DBIT -> refers to Refunds / Disputes. Disputes flow is yet to be added to marked Refunds as default case.

Reply:
`trustpay::CreditDebitIndicator::Crdt` use variant instead of _ (discarding are variant cause grey failures)